### PR TITLE
Set parent for VM's where possible

### PR DESCRIPTION
**Pull request by djberg96**
This PR sets the parent image for virtual machines in Azure. Previously, this was only getting set via provisioning within the tool.

Note that this does not currently support marketplace images. Setting this is not impossible, but will require a change in how we collect and/or store marketplace image data. This is because Azure always sets the version to "latest" for a VM, whereas there can be many different versions for the same markeplace image, and more are added over time.

Addresses https://bugzilla.redhat.com/show_bug.cgi?id=1478889

~~WIP for now until targeted refresh is fixed and the specs pass.~~

### CODE REVIEW

**Comment by djberg96** [comment]
@Ladas @tumido I need some help here. I'm not sure how to set the parent for targeted refresh.

File: app/models/manageiq/providers/azure/inventory/parser/cloud_manager.rb [hunk 1]
```diff
@@ -83,6 +84,7 @@ def instances
         :raw_power_state     => status,
         :flavor              => series,
         :location            => instance.location,
+        :parent              => persister.vms.lazy_find(parent_ref),
```
Comment (Ladas) [inline app/models/manageiq/providers/azure/inventory/parser/cloud_manager.rb]: it looks like the parent is an Image? So this shouldbe correct:
```ruby
:parent              => persister.miq_templates.lazy_find(parent_ref),
```

Comment (Ladas) [reply app/models/manageiq/providers/azure/inventory/parser/cloud_manager.rb]: this is supposed to be image used for provisioning of the Vm, right?

**Comment by djberg96** [comment]
@Ladas Ok, made that change. Now there's all sorts of weird failures. :(

File: app/models/manageiq/providers/azure/inventory/parser/cloud_manager.rb [hunk 2]
```diff
@@ -83,6 +84,7 @@ def instances
         :raw_power_state     => status,
         :flavor              => series,
         :location            => instance.location,
+        :parent              => persister.miq_templates.lazy_find(parent_ref),
```
Comment (Ladas) [inline app/models/manageiq/providers/azure/inventory/parser/cloud_manager.rb]: it has to be called ```genealogy_parent```, there is after save logic handling this as backup, but we use custom saver saving this

**Pull request merged**